### DISCUSSION TITLE
Use Thanos Store as backend of Mimir Querier

### DISCUSSION THREAD

**Discussion by jobinjosem1** [grafana/mimir#8520] [Help and support]
I am in the process of setting up Mimir and need to migrate historical data stored by Thanos in a storage account. I was wondering if it’s possible to configure the Thanos store as a backend for the Mimir querier, allowing it to directly query Thanos for historical metrics without needing to convert the existing data. Has anyone implemented this setup before? If so, could you provide guidance on the necessary configuration changes? Your assistance would be greatly appreciated.

**Comment by dimitarvdimitrov** [comment]
That won't be possible I'm afraid. The mimir querier discovers the store-gateways (Mimir's equivalent of Thanos' store) via the hash ring. Since Thanos does not maintain that hash ring, it's not possible to connect the two.
